### DISCUSSION TITLE
Catch idleTimeout exception in scala play

### DISCUSSION THREAD

**Discussion by mrcvtl** [playframework/playframework#11428] [Q&A]
Hi guys, I've been trying to manage idleTimeout in my microservice  (scala play 2.6.x) for a couple of days with no results.

First of all, i set the play.server.http.idleTimeout = 3s and the equivalent in the build.sbt as a PlayKeys.devSettings. The timeout works well but returns a generic connection error on the client side. To try not to blow up the business service that calls the above microservice, I would like the event that causes the idleTimeout (clearly an exception) to be handled with a catch, but I cannot understand where this exception occurs.
I try using a try and catch into the GET method, outside of it (outside the Action)  i try the .recover using an Action.async and a Future, and all of this catching every Throwable with absolutely no results.

Anyone can help me?

**Comment by jducoeur** [comment]
Hmm.  I don't think I've used idleTimeout like this, but why do you think it's a server-side exception?  I would have guessed that the framework itself is shutting down the connection, and the error is purely client-side -- that it's not coping with the relatively abrupt shutdown.
